src/app/product-list/product-list.component.ts

```diff
@@ -12,15 +17,14 @@ import { map } from 'rxjs/operators';
 export class ProductListComponent implements OnInit {
   products$: Observable<{ product: Product; isInCart: boolean }[]>;
 
-  constructor(
-    private productService: ProductService,
-    private cartService: CartService
-  ) {}
+  constructor(private store: Store<AppState>) {}
 
   ngOnInit() {
+    this.store.dispatch(loadProducts());
```

Comment:
do not load product here... read about effect life cycle and load products there